# HW 2

Just generic python setup first, including some helper functions to help with degree-radians conversion, and solving quadratic equations

In [1]:
import numpy as np
from numpy import sin, cos, tan, arctan2, sqrt

In [2]:
def rad(th):
    return th * np.pi/180

def deg(th):
    return th * 180/np.pi

def quadratic(a, b, c):
    return ((-b + sqrt(b**2 - 4*a*c)) / (2*a),
            (-b - sqrt(b**2 - 4*a*c)) / (2*a))

## Problem 5

There's already a sketch that comes with the problem, so I'm using that.

We know $\theta = 60^\circ$, $v_0 = 40$ m/s, $t_f = 5.5$ s, and the acceleration due to gravity.

In [3]:
th = rad(60.)
v0 = 40.
tf = 5.5
g = 9.8

We'll get to use most of our projectile motion equations:
\begin{align}
x(t) &= v_{0x} t + x_0\\
y(t) &= -\frac{1}{2}gt^2 + v_{0y} t + y_0\\
v_x(t) &= v_{0x}\\
v_y(t) &= -gt + v_{0y}
\end{align}

Putting the origin of the coordinate system at the launch point, $x_0 = y_0 = 0$. Now we can find where the rock ends up when it lands at $t_f$. We first have split the initial velocity into components, though.

In [4]:
x0 = 0
y0 = 0
v0x = v0 * cos(th)
v0y = v0 * sin(th)
def x(t): return v0x * t + y0
def y(t): return -.5 * g * t**2 + v0y * t + y0

print("(x_f, y_f) = ({:f}, {:f})".format(x(tf), y(tf)))

(x_f, y_f) = (110.000000, 42.300589)


So we find $H = 42.3$m and $R = 110$ m. Let's find how fast it's going first:

In [5]:
def vy(t): return -g*t + v0y

vfx, vfy = v0x, vy(tf)
vf = sqrt(vfx**2 + vfy**2)

print("vf = ({:f}, {:f}) magnitude = {:f}".format(vfx, vfy, vf))

vf = (20.000000, -19.258984) magnitude = 27.765238


So it's hitting the ground going 28 m/s.

Finally, to get the maximum height, we can use the $y(t)$ formula, but we first need to find the proper time $t_{peak}$ when the stone reaches peak height. Realizing that before reaching peak height the stone is moving up, while afterwards it's moving down, we can use that right at the peak it's moving neither up nor down.

\begin{align}
0 & = v_y(t_{peak})\\
 & = -g t_{peak} + v_{0y}\\
\end{align}

We can solve this to find
$$
t_{peak} = \frac{v_{0y}}{g}
$$

and plugging that time back into $y(t)$ we find the peak height $y_{max}$:

\begin{align}
y_{max} &= y(t_{peak})\\
&= -\frac{1}{2}g\left(\frac{v_{0y}}{g}\right)^2 + v_{0y} \frac{v_{y0}}{g} + y_0\\
&= \frac{v_{0y}^2}{2g} + y_0\\
\end{align}

Plug in numbers:

In [6]:
ymax = v0y**2 / (2*g) + y0
print("ymax = {:f}".format(ymax))

ymax = 61.224490


The rock reaches a peak height of 61 m.